examples/platforms/simulation/platform-config.h

```diff
@@ -116,3 +116,13 @@
 #ifndef OPENTHREAD_SIMULATION_MDNS_SOCKET_IMPLEMENT_POSIX
 #define OPENTHREAD_SIMULATION_MDNS_SOCKET_IMPLEMENT_POSIX 0
 #endif
+
+/**
+ * @def OPENTHREAD_SIMULATION_IMPLMENT_INFRA_IF
+ *
+ * Define as 1 for the simulation platform to provide implementation of `otPlatInfra` APIs.
+ *
+ */
+#ifndef OPENTHREAD_SIMULATION_IMPLEMENT_INFRA_IF
+#define OPENTHREAD_SIMULATION_IMPLEMENT_INFRA_IF 0
```

Comment:
How about making the default value `1`? The case where we don't want it provided seems to be the more specialized scenario.